src/main/scala/algolia/definitions/WaitForTaskDefinition.scala

```diff
@@ -64,6 +60,19 @@ trait WaitForTaskDsl {
   implicit object WaitForTaskDefinitionExecutable
       extends Executable[WaitForTaskDefinition, TaskStatus] {
 
+    // Run every 50 ms, use a wheel with 512 buckets
+    private lazy val timer = {
+      val threadFactory = new ThreadFactory {
```

Comment:
couldn't we reuse the thread pool from the execution context?
Not a big fan of hiding a theadFactory.